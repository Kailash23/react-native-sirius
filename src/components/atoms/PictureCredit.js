import React from 'react';
import { View } from 'react-native';
import { withTheme, Caption, Divider } from 'react-native-paper';
import { SIZES } from '@styles/sizes';
import { customFonts } from '@styles/customFonts';

function PictureCredit(props) {
  const { credit, theme, style } = props;
  return (
    credit && (
      <View style={{ marginVertical: 5, marginBottom: SIZES.PADDING * 2 }}>
        <Caption
          style={[style, { fontFamily: customFonts.merriweatherSansRegular }]}>
          {credit}
        </Caption>
        <Divider style={[styles.divider, { borderColor: theme.colors.line }]} />
      </View>
    )
  );
}

const styles = {
  divider: {
    borderWidth: 0.5
  }
};
export default withTheme(PictureCredit);
