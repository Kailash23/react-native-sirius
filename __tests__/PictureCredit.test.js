import PictureCredit from '@atoms/PictureCredit';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

configure({ adapter: new Adapter() });

describe('Testing <PictureCredit/>', () => {
  test('PictureCredit should render properly with given Credit text', () => {
    const wrapper = shallow(<PictureCredit credit={'Test picture credit'} />);
    expect(wrapper).toMatchSnapshot();
  });
  test('PictureCredit should render nothing if no credit text is passed or is null', () => {
    const wrapper = shallow(<PictureCredit credit={null} />);
    expect(wrapper).toMatchSnapshot();
  });
});
