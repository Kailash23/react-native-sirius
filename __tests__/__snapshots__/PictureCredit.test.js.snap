// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <PictureCredit/> PictureCredit should render nothing if no credit text is passed or is null 1`] = `
<ThemedComponent
  _reactThemeProviderForwardedRef={null}
  credit={null}
/>
`;

exports[`Testing <PictureCredit/> PictureCredit should render properly with given Credit text 1`] = `
<ThemedComponent
  _reactThemeProviderForwardedRef={null}
  credit="Test picture credit"
/>
`;
